support/oidc-discovery-provider/main.go

```diff
@@ -91,10 +97,19 @@ func run(configPath string) error {
 		Handler:           handler,
 		ReadHeaderTimeout: 10 * time.Second,
 	}
+
+	go func() {
+		<-ctx.Done()
+		err = server.Shutdown(context.Background())
+		if err != nil {
```

Comment:
In general, it isn't safe to modify the `err` variable from the goroutine. In practice in this code it is fine because it isn't used after this point. It would be safer to use a new `err` variable, ideally scoped to the shutdown statement:
```
if err := server.Shutdown(context.Background()); err != nil {
    log.Error(err)
}
```